src/opnsense/scripts/shell/setaddr.php

```diff
@@ -259,26 +269,28 @@ function add_gateway_to_config($interface, $gatewayip, $inet_type, $is_in_subnet
         echo "\n";
     }
 
-    if ($new_name == '') {
+    if (empty($new_name)) {
         $new_name = next_unused_gateway_name($interface);
     }
 
-    $item = array(
+    $item = [
+        'disabled' => 0,
         'descr' => sprintf('Interface %s Gateway', strtoupper($interface)),
-        'defaultgw' => $is_default,
+        'defaultgw' => $is_default ? 1 : 0,
         'ipprotocol' => $inet_type,
         'interface' => $interface,
         'gateway' => $gatewayip,
         'monitor_disable' => 1,
         'name' => $new_name,
-        'interval' => true,
         'weight' => 1,
-    );
-    if (!$is_in_subnet) {
-        $item['fargw'] = 1;
-    }
-
-    $a_gateways[] = $item;
+        'fargw' => !$is_in_subnet ? 1 : 0
+    ];
+
+    $gw->createOrUpdateGateway($item, $uuid);
+    /* Serialize back to global config, capturing the changes made here.
```

Comment:
```suggestion
    /* 
     * Serialize back to global config, capturing the changes made here.
```

multiline comments be like this :D